### PR TITLE
Add verify-docker-build to pre-submit tests

### LINKED ISSUE #60: Add verify script to run `docker build`
/kind feature

**Describe the solution you'd like**
I'd like the verify scripts to also build the docker image (not push, just build and make sure it's working)

### PR DESCRIPTION
**Pull request by ashish-amarnath**
**What this PR does / why we need it**:
To ensure an image for `capd-manager` can be built with the proposed changes in a PR

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #60 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ashish-amarnath** [comment]
We need docker in docker enabled for the pre-submit job to be able to do this

**Comment by ashish-amarnath** [comment]
https://github.com/kubernetes/test-infra/pull/13366 should update the pre-submit job config.

File: hack/verify-docker-build.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# check if manager docker image builds
+cd_root_path
+
+# DEBUG CODE BEGIN
+service docker start
+# the service can be started but the docker socket not ready, wait for ready
+WAIT_N=0
+MAX_WAIT=5
+while true; do
```
Comment (chuckha) [inline hack/verify-docker-build.sh]: This isn't part of the test runner image?

Comment (chuckha) [reply hack/verify-docker-build.sh]: ah it was taken from here

https://github.com/kubernetes/test-infra/blob/master/images/bootstrap/runner.sh#L84

Comment (chuckha) [reply hack/verify-docker-build.sh]: unless i'm missing something we don't need to do that in our verify script

Comment (ashish-amarnath) [reply hack/verify-docker-build.sh]: this was a debug commit to see if docker binaries are built into this image we are using. This PR needs https://github.com/kubernetes/test-infra/pull/13366

Comment (ashish-amarnath) [reply hack/verify-docker-build.sh]: debug code was removed. Resolving this.

**Comment by chuckha** [comment]
/approve

thanks !

**Pull request merged**